### ISSUE TITLE
build fails with musl libc (e.g. alpine linux)

### ISSUE THREAD

**Issue by rofl0r** [DynamoRIO/dynamorio#2324]
```
dynamorio-release_7_0_0_rc1/core/unix/os.c:6087:30: error: dereferencing pointer to incomplete type 'stdfile_t {aka struct _IO_FILE}'
             (*privmod_stdout)->STDFILE_FILENO = our_stdout;
                              ^~
make[2]: *** [core/CMakeFiles/dynamorio_static.dir/unix/os.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/src/build/dynamorio/dynamorio-release_7_0_0_rc1/build'
make[1]: *** [core/CMakeFiles/dynamorio_static.dir/all] Error 2
make[1]: Leaving directory `/src/build/dynamorio/dynamorio-release_7_0_0_rc1/build'
make: *** [all] Error 2
```

**Comment by rofl0r** [comment]
after that's fixed by commenting out the non-portable manipulation of an opaque C type...

```
In file included from /usr/include/signal.h:48:0,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/lib/globals_shared.h:126,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/../globals.h:90,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal_private.h:51,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:45:
/usr/include/bits/signal.h:37:16: error: redefinition of 'struct _fpstate'
 typedef struct _fpstate {
                ^~~~~~~~
In file included from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal_private.h:49:0,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:45:
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/include/sigcontext.h:157:8: note: originally defined here
 struct _fpstate {
        ^~~~~~~~
In file included from /usr/include/signal.h:48:0,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/lib/globals_shared.h:126,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/../globals.h:90,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal_private.h:51,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:45:
/usr/include/bits/signal.h:49:8: error: redefinition of 'struct sigcontext'
 struct sigcontext {
        ^~~~~~~~~~
In file included from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal_private.h:49:0,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:45:
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/include/sigcontext.h:180:8: note: originally defined here
 struct sigcontext {
        ^~~~~~~~~~
In file included from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/../globals.h:90:,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal_private.h:51,
                 from /src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:45:
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal_private.h: In function 'libc_sigismember':
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal_private.h:519:45: error: 'sigset_t {aka const struct __sigset_t}' has no member named '__val'
     return TEST(1UL << (sig % bits_per), set->__val[sig / bits_per]);
                                             ^
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/lib/globals_shared.h:435:40: note: in definition of macro 'TESTANY'
 #define TESTANY(mask, var) (((mask) & (var)) != 0)
                                        ^~~
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal_private.h:519:12: note: in expansion of macro 'TEST'
     return TEST(1UL << (sig % bits_per), set->__val[sig / bits_per]);
            ^~~~
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c: In function 'thread_set_self_context':
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:2375:34: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
     frame.uc.uc_mcontext.fpstate = &xstate.fpstate;
                                  ^
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c: In function 'execute_handler_from_cache':
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:4844:44: error: 'SA_NOMASK' undeclared (first use in this function)
     if ((info->app_sigaction[sig]->flags & SA_NOMASK) == 0)
                                            ^~~~~~~~~
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:4844:44: note: each undeclared identifier is reported only once for each function it appears in
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:4885:44: error: 'SA_ONESHOT' undeclared (first use in this function)
     if ((info->app_sigaction[sig]->flags & SA_ONESHOT) != 0) {
                                            ^~~~~~~~~~
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c: In function 'execute_handler_from_dispatch':
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:5053:44: error: 'SA_NOMASK' undeclared (first use in this function)
     if ((info->app_sigaction[sig]->flags & SA_NOMASK) == 0)
                                            ^~~~~~~~~
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:5091:44: error: 'SA_ONESHOT' undeclared (first use in this function)
     if ((info->app_sigaction[sig]->flags & SA_ONESHOT) != 0) {
                                            ^~~~~~~~~~
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c: In function 'execute_default_action':
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:5204:44: error: 'SA_ONESHOT' undeclared (first use in this function)
         (info->app_sigaction[sig]->flags & SA_ONESHOT) != 0) {
                                            ^~~~~~~~~~
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c: In function 'handle_sigretur':
/src/build/dynamorio/dynamorio-release_7_0_0_rc1/core/unix/signal.c:5594:44: error: 'SA_ONESHOT' undeclared (first use in this function)
     if ((info->app_sigaction[sig]->flags & SA_ONESHOT) != 0) {
                                            ^~~~~~~~~~
cc1: all warnings being treated as errors
```

**Comment by derekbruening** [comment]
Looks like a duplicate of #1973
